### DISCUSSION TITLE
表单验证提示语如何设置多语言的语言包？

### DISCUSSION THREAD

**Discussion by rojer95** [DouyinFE/semi-design#2250] [Q&A]
表单验证提示语如何设置多语言的语言包？
---
表单验证是基于`async-validator`的，
默认在`rules`中未配置`message`情况下是显示默认的英文提示语（这个是`async-validator`内置的）

`Form`组件有没有什么方式给`async-validator`设置语言包，或者在Semi Design的`LocaleProvider `组件中进行注入语言包？

（`async-validator`是支持设置语言包的：https://github.com/yiminghe/async-validator?tab=readme-ov-file#messages）

**Comment by pointhalo** [comment]
这个自己传一下 message就好了。

你说的asynca-validator支持设置语言包是指这里最后这种方式吗，它也不是切换locale的形式呀。也是你自己传进去的。跟message里直接传没有区别。
```
validator.messages(cn);
```
![image](https://github.com/DouyinFE/semi-design/assets/88709023/6d2eacbe-f5d3-4a14-9f7f-5a9f89395e99)

**Reply by pointhalo** [reply]
校验信息都是高度灵活的，这个地方内置 i18n的实用价值不高。

**Reply by white1022** [reply]
主要是不想每个都写验证消息，直接用它默认的 但是默认的又不是中文，就很难受...

**Comment by rojer95** [comment]
![image](https://github.com/DouyinFE/semi-design/assets/20662049/8addc167-b507-4178-a5d2-8bf6b8043ca7)

对于比如长度验证的提示语，这种都是很通用的提示语，完全可以语言包，

`async-validator`的`fullField`支持对Label进行替换的

以上是我的个人见解
